server/src/main/java/org/elasticsearch/action/ActionListener.java

```diff
@@ -301,9 +265,55 @@ public String toString() {
         };
     }
 
+    /**
+     * @deprecated in favour of {@link #running(Runnable)} because this implementation doesn't "wrap" exceptions from {@link #onResponse}
+     * into {@link #onFailure}.
+     */
+    @Deprecated(forRemoval = true)
+    static <Response> ActionListener<Response> wrap(Runnable runnable) {
+        return running(runnable);
+    }
```

Comment:
I am puzzled by this. Since it is delegating all calls to `running(..)` means it is the same and could that be in-lined and removed now?

Reply:
I think we use this in the stateless codebase too, so by leaving it here for now we don't break that.